### DISCUSSION TITLE
How to view multiple shapes in replicad-app-example

### DISCUSSION THREAD

**Discussion by beau-haldane** [sgenoud/replicad#143] [Q&A] [answered]
Hey @sgenoud! Firstly, Replicad is an incredible tool - it is enabling me to complete a project I've been thinking about for a very long time, so thank you :)

I would like for my app to behave very similarly to [Deck in a Box](https://deckinabox.sgenoud.com/quick-draw) where users can change model parameters and view/download the **multiple** associated files required to print the model. I may be missing something incredibly obvious, but I cannot seem to figure out how to display multiple models in my app which is based off `replicad-app-example`. I have had no issues getting them to display in Replicad Studio.

[Here's my project repo](https://github.com/beau-haldane/3d-printed-snare-drum) - I managed to get the user inputs generating different versions of the model relatively easily - I just can't display the full 'assembly' of the model. For reference, [here's my model in Replicad Studio](https://studio.replicad.xyz/workbench#code=UEsDBAoAAAAIAEUJUlheRcbDWgYAAKweAAAHAAAAY29kZS5qc9VZbU%252FjOBD%252B3l9hre5DCmlpy7Laaw9OvBwH0q2ooLrlFiFtSN0mIk2q2NkFof73GzuOYztOoEWsdvlQUtsz88z7OPWTmFD0hKap993ln5dJFk%252Fx9BL71IvnEc5Xj8PUjzBaoX2U4mUU%252Bt501Gr5nHrppd4CU5wS2H1qIRRlczJEH114nIb51nnsBxgW%252BwO%252Bipc0KJbesxUS4CiaBKF%252FH2MCi7%252B7OZ%252BzJMIngscQ7SmrV0vPD%252BP5EH3gq7OQLnBMJ7CTerGPh6jXBVkrQLmzg85wtMQpOs1in4YAWiAPOYRJ8imMolCq4LDlNto%252F4PtoCw32uu%252Bltr4X%252BVnkUXz9HzsbPC4TiuOM4H9wPKeBi7jVrjk9M0ZOxRcvPTBHzEXkh4D3J48G3fF5G%252B2g%252FsfeSBI8wCmTd3GchLGjMmyXZI8NZH5CTDKgSzHN0hjdPLjo8XbEDWYo%252Bi9OKX44ZJQMegrUGXHR93BKA8VyhsYg7Rjtw1eEnAFAyMnKh06FAexVee7UclC0NqzLtfWYugyEee4Ez1OM2TmNbAs54ABwQ%252BER1ToqoWqjOY4h3Ci%252BwnMWfUfJlDnASTIAfyns9K00n4toEeJungTSaEIQyzWnzXXuLpJveJyEMfCaJI4Sd7X8Af2gLahpAGA5OTlM%252FSYGHZODi244D4R6rnhQCfjSrRAThTGuZw5eVjSuCFoTqs5tPdQaraaBHyUEFzYn95j6wUU8jrxYLuIHmmZT7OQes%252Fn%252FFEIWAnZqxEGrYjg1GOCrBomzb%252F3yEVERIwg2CxHXQi2Wav3LCF15qFOeR6Zp0LayZyt4dUi5CP1rW7IqnvL%252FP0WQr%252BkUi%252BBX%252B%252BEHmF4z%252BGZ5DcPCSZot0N95Ziepkem7J%252BOrE5bbT5XpxphsxKxCLJNMdYyxzUCaK835BkYbWSkA8zED6YF3wVJXjI%252FseRwRALbMOo6CVumSos%252FWkGgqs7FlUFJyDUQN1KeG3Q%252BstTJz8LZqID7CXgpGQH9N55hIbnf5Kls8w%252BE8oIyPMiIp%252Bw0yHSaUB4lN8sS7KwVS7%252B4zG0CAtq%252FIgWWdsTVai7GoYNLWGEi3FuPJLEqS1FGjeFenuFCif78cmhSSjsbYcAVsnRohUw4%252BVgV4TqRKQTdDDhYViLwLRYl%252FDy4AK0oPMcvJQ3I%252BUgIjn7A14jE8YVpOkLZ4OvIIs71l4irqkIod1cVisW3UyaL78vrBFbBhAKjNEHS%252FueWqDYHiPAuIbppQEOJ0nLqs4kXUjE7e75sUyG29jhogpFp9bJrlaOrCbqtsBlpCVLmrRy0WabJHx2YPzRyi7oIBynutY9RntacYrePd9Zd3MKaIdAfwe215sugnVNGt36v3BmAwI7xgNYPbm2N4TsrxM2ru5V7VTsj2KpRT2i3rjbINiqVNjavyoOA%252BwurKl7y1CCq91VUb9k8GePttAXd%252BORO%252FAeJGG9cmzOeQBmW74BOKrY90UzzNfFxoq4UIb9AiW8p2w%252F7EfUs9zG1Wnh%252BJsytbj9Fah4WXjp2xWlka5bmuEH9XVGmLgIiMcVpWEX0oQwfQhQfoTzDzEPUFIph7xEiLi6GnppMrrcFyrW3udPWBwt%252Fg6MDbTZ3QOl5Y2rM1AjZRoaHLraeVcrGp74YbdMLnLcJmW5bEyIunaJyQkL3rZN5GoHaIG3w%252BSZbVbJKFQysM1Xzu1MyCldZrHDtKKE0Wm8jtmAOkIvEF4WHVNt96a51zKXWavxTDS%252FRXE16tKs9kP6s4N7ej5zRQj82SFG5ZkHUhLPZG8O8PftOrpAZsbW%252B3ZckVt03WYyBOL2YzwjJXTR61yh2gPpS0hlQ0zkPmtKH%252B9UZWaeIqVxZ%252FHcU2KLGFGmYLI%252BdVIZC73SiZO5ooecI0d3eZkcApuxAJvCUeWvKzCIeiKevsy9csMbSCIfpquhz99lQRHeU%252FD8CQuvpaMFi9Emke2j8e7AwulRB3Zi0u4daMClwpZVypKxjNGkkGEtDrBL7EilWZq1b%252BqY4gTzU43Grir%252BqGkkNC8OIuejSGkZdWmHUgyJSsG4gcW33jrxeQOohZL1u2txF1ryJKKUJ7XPAnpgDFPM%252BVRKMCauQ1%252BaXPo3pAsCJVU6OqSaYZiWWYPZf08LnpdrsaShfBium32zJyBJ0ZOo72s8nCC2MtmAwq%252FpLVKX9Xzon%252FB1BLAQIUAAoAAAAIAEUJUlheRcbDWgYAAKweAAAHAAAAAAAAAAAAAAAAAAAAAABjb2RlLmpzUEsFBgAAAAABAAEANQAAAH8GAAAAAA%253D%253D).

Thank you for any assistance!

**Comment by sgenoud** [comment, marked as answer]
Nice looking thing, I want photos when it is printed! 😉 

I depends on how you want to show it! First you can look at [how it is done in the studio](https://github.com/sgenoud/replicad/blob/main/packages/studio/src/ShapeGeometry.jsx#L129). What I do is I take an array of models and show them (with a simple map).

You could also merge all the models as one with `makeCompound` and just have one model to show.

Does it help?

**Reply by beau-haldane** [reply]
Yes it works! Thank you!

![image](https://github.com/sgenoud/replicad/assets/62363339/994ccfb8-73e3-41fb-b9db-eacc59c18ac8)
